lib/Doctrine/ODM/MongoDB/Types/DateType.php

```
@@ -69,13 +66,15 @@ public static function getDateTime($value) : DateTimeInterface
         return $datetime;
     }
 
-    private static function craftDateTime(int $seconds, $microseconds = 0) : DateTime
+    /**
+     * @return DateTime|false
```

I think we should throw an exception on `false` and get back to returning `\DateTime`

---

We do this already a few lines later. The reason we don't do this from within `craftDateTime` is that we want to display the raw value that couldn't be converted, which isn't available inside.